pandas/core/frame.py

```diff
@@ -4621,9 +4624,15 @@ def drop_duplicates(
         if inplace:
             (inds,) = (-duplicated)._ndarray_values.nonzero()
             new_data = self._data.take(inds)
+
+            if ignore_index:
```

Comment:
Giving this some more thought - does it even make sense to accept this keyword with `inplace`? By definition `inplace` should match the existing structure so I think should raise of both `inplace` and `ignore_index` are supplied, unless I'm missing something

Reply:
emm, I get your point, and i think it still makes sense to accept this keyword with `inplace`, because the definition of `inplace` is `Whether to drop duplicates in place or to return a copy`, and if there is duplicates, the structure is anyway sort of changed (since index will change although the order is not). so if people want to reset the order, they still have to do the resetting themselves afterwards.

Reply:
Yea valid point as well; generally I think the semantics here are a little murky. Let's see what @jreback thinks

Reply:
Yea valid point as well; generally I think the semantics here are a little murky. Let's see what @jreback thinks

Reply:
Yea valid point as well; generally I think the semantics here are a little murky. Let's see what @jreback thinks